api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -81,6 +81,7 @@ func (mw *azureManagedControlPlaneWebhook) Default(ctx context.Context, obj runt
 	setDefault[*Identity](&m.Spec.Identity, &Identity{
 		Type: ManagedControlPlaneIdentityTypeSystemAssigned,
 	})
+	setDefault[*bool](&m.Spec.EnablePreviewFeatures, ptr.To(false))
```

Comment:
For next time, the compiler can usually figure out what the type parameters are so they don't always need to be included.
```suggestion
	setDefault(&m.Spec.EnablePreviewFeatures, ptr.To(false))
```